### ISSUE TITLE
undefined reference to `bpf_program__attach_usdt'

### ISSUE THREAD

**Issue by Rtoax** [iovisor/bcc#4222]
```
$ cd bcc/libbpf-tools
$ make
```

```
  GEN-SKEL hardirqs.skel.h
  CC       hardirqs.o
  BINARY   hardirqs
  BPF      javagc.bpf.o
  GEN-SKEL javagc.skel.h
  CC       javagc.o
javagc.c: In function ‘main’:
javagc.c:190:48: warning: implicit declaration of function ‘bpf_program__attach_usdt’; did you mean ‘bpf_program__attach_xdp’? [-Wimplicit-function-declaration]
  190 |         skel->links.handle_mem_pool_gc_start = bpf_program__attach_usdt(skel->progs.handle_gc_start, env.pid,
      |                                                ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                                bpf_program__attach_xdp
javagc.c:190:46: warning: assignment to ‘struct bpf_link *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  190 |         skel->links.handle_mem_pool_gc_start = bpf_program__attach_usdt(skel->progs.handle_gc_start, env.pid,
      |                                              ^
javagc.c:198:44: warning: assignment to ‘struct bpf_link *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  198 |         skel->links.handle_mem_pool_gc_end = bpf_program__attach_usdt(skel->progs.handle_gc_end, env.pid,
      |                                            ^
javagc.c:206:37: warning: assignment to ‘struct bpf_link *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  206 |         skel->links.handle_gc_start = bpf_program__attach_usdt(skel->progs.handle_gc_start, env.pid,
      |                                     ^
javagc.c:214:35: warning: assignment to ‘struct bpf_link *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  214 |         skel->links.handle_gc_end = bpf_program__attach_usdt(skel->progs.handle_gc_end, env.pid,
      |                                   ^
  BINARY   javagc
/usr/bin/ld: /home/rongtao/Git/rtoax/bcc/libbpf-tools/.output/javagc.o: in function `main':
/home/rongtao/Git/rtoax/bcc/libbpf-tools/javagc.c:190: undefined reference to `bpf_program__attach_usdt'
/usr/bin/ld: /home/rongtao/Git/rtoax/bcc/libbpf-tools/javagc.c:198: undefined reference to `bpf_program__attach_usdt'
/usr/bin/ld: /home/rongtao/Git/rtoax/bcc/libbpf-tools/javagc.c:206: undefined reference to `bpf_program__attach_usdt'
/usr/bin/ld: /home/rongtao/Git/rtoax/bcc/libbpf-tools/javagc.c:214: undefined reference to `bpf_program__attach_usdt'
collect2: error: ld returned 1 exit status
make: *** [Makefile:168: javagc] Error 1
```

Need libbpf code commit 1b4b798916b3eb0da6a149343fa6b5deabf74517 at least.

**Comment by chantra** [comment]
CI was showing failure to include so usdt.h file earlier on fedora runners. That could be related.

**Comment by chenhengqi** [comment]
@Rtoax  Looks like the submodule is not up to date. Try `git submodule update --init --recursive`.

**Comment by chenhengqi** [comment]
I see, #4186 rolled back libbpf submodule.

**Comment by Rtoax** [comment]
Yes, i think so.

**Comment by chenhengqi** [comment]
Addressed by #4224.
